### ISSUE TITLE
White Lines Appearing & Disappaaring All Over GUI (Buggy Element Highlighting?)

### ISSUE THREAD

**Issue by Stargrazer82301** [spyder-ide/spyder#16230]
### Issue Report Checklist

* [X] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [X Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [X] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [X] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [X] Tried basic troubleshooting (if a bug/error)
    * [X] Restarted Spyder
    * [X] Reset preferences with ``spyder --reset``
    * [X] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [X] Tried the other applicable steps from the Troubleshooting Guide
* [X] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description

There are lots of little white lines that appear & disappear between GUI elements, as I move my mouse over the screen. It looks like perhaps some part of the highlighting visual effect is buggy? I have attached a couple of screenshots, with the little white lines highlighted. They're not not super obvious in the screenshots, but they are very distracting as dozens of them they continually appear & disappear all over the window as I move my mouse around the GUI.

I am using Spyder 5.1.1 on Mac Catalina.

### What steps reproduce the problem?

1. Move mouse over GUI elements

### What is the expected output? What do you see instead?

 I would expect no little white lines to appear over GUI elements. As mentioned it seems like something is going wrong about how GUI elements are highlighted 

## Versions

* Spyder version: 5.1.1
* Python version: 3.7.1 64-Bit
* Qt version: 5.12.10
* PyQt version: 5.12.3
* Operating System name/version: MacOS Catalina 10.15.7

### Dependencies
<img width="2399" alt="Screenshot 2021-08-17 at 12 41 26" src="https://user-images.githubusercontent.com/2066338/129770516-27b73b27-00f7-4df7-a265-6dd480d359d1.png">
<img width="1065" alt="Screenshot 2021-08-17 at 12 44 45" src="https://user-images.githubusercontent.com/2066338/129770518-44df1db1-ddbb-415d-93d8-84d583c38acb.png">

# Mandatory:
applaunchservices >=0.1.7     :  0.2.1 (OK)
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  1.6.0 (OK)
cookiecutter >=1.6.0          :  1.7.2 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.6.0               :  7.24.1 (OK)
jedi >=0.17.2;<0.19.0         :  0.17.2 (OK)
jsonschema >=3.2.0            :  3.2.0 (OK)
keyring >=17.0.0              :  23.0.1 (OK)
nbconvert >=4.0               :  6.0.7 (OK)
numpydoc >=0.6.0              :  1.1.0 (OK)
parso >=0.7.0;<0.9.0          :  0.7.0 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.8.0 (OK)
pygments >=2.0                :  2.9.0 (OK)
pylint >=2.5.0                :  2.8.3 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.2.1;<1.3.0          :  1.2.1 (OK)
pylsp_black >=1.0.0           :  None (OK)
qdarkstyle =3.0.2             :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.1.10 (OK)
qtawesome >=1.0.2             :  1.0.2 (OK)
qtconsole >=5.1.0             :  5.1.0 (OK)
qtpy >=1.5.0                  :  1.9.0 (OK)
rtree >=0.9.7                 :  0.9.7 (OK)
setuptools >=49.6.0           :  57.0.0 (OK)
sphinx >=0.6.6                :  4.0.2 (OK)
spyder_kernels >=2.1.0;<2.2.0 :  2.1.0 (OK)
textdistance >=4.2.0          :  4.2.1 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  1.0.2 (OK)
zmq >=17                      :  22.1.0 (OK)

# Optional:
cython >=0.21                 :  0.29.15 (OK)
matplotlib >=2.0.0            :  3.3.4 (OK)
numpy >=1.7                   :  1.17.3 (OK)
pandas >=1.1.1                :  0.25.3 (NOK)
scipy >=0.17.0                :  1.6.0 (OK)
sympy >=0.7.3                 :  None (OK)

**Comment by ccordoba12** [comment]
Hey @Stargrazer82301, thanks for reporting. This could be caused because you're using a very high screen resolution.

Could you let us know what's your current screen resolution to check that?

**Comment by Stargrazer82301** [comment]
Hi; it is indeed a high-resolution screen, 5120x2880. Is there a Spyder setting I can change to remedy it?

**Comment by Stargrazer82301** [comment]
PS: Just checked, and I am also experiencing it on a 2560x1600 screen.

**Comment by ccordoba12** [comment]
> Is there a Spyder setting I can change to remedy it?

No, there isn't, sorry. Also, I don't know how to fix this problem.

**Comment by tony-pythony** [comment]
I made a similar observation, except I've a 1080p screen & resolution set. 
When I use auto-dpi scaling, everything is fine, but the UI is to small for my old eyes and my big mouse.
So, scaling spyder's UI to a ration of 1.12 makes working comfortable possible. But, I see, too, lines between Buttons (sometimes blinking irregularly), On texts, it looks like font-hinting isn't working properly (the part with window's ClearType also shows the same effect on ordinary windows software/texts/etc.. )
so I think it's the (auto) high-dpi scaling of Spyder leading to this. May be the scaling ratio is miscalculated somehow?

In my case setting the dpi-scaling to 1.125 (instead of 1.12) makes that effect go away.
May be its good work around to try different scalings like 1.05 or 99.995??? 
But, yes.. its a kind of a bug with the rendering system.

Hope that helps a little bit, my friends.
Thank you for investing you time & knowledge!

**Comment by Stargrazer82301** [comment]
Interesting. To test, I just tried setting my DPI scaling to 1.125, but I'm still seeing the lines. Odd that it should vary between systems like this.
